### PR TITLE
Take into account height of the hat for hat blocks.

### LINKED ISSUE #330: workspace metrics incorrect when top block has hat
It seems that since https://github.com/google/blockly/commit/a67f841671dd5d61647ea4f159350e5823b69e69 the computation of a workspace metric does not take into account hats.

repro: create a workspace with a single hat block, measure size and compare to actual SVG size reported by the debugger.

### PR DESCRIPTION
**Pull request by picklesrus**
Also, position hat blocks correctly in the flyout/toolbox.
This fixes #330.

### CODE REVIEW

**Comment by picklesrus** [comment]
@NeilFraser I don't love this fix, but maybe it is the right way?  It seems to be similar to how the tab width is dealt with for the puzzle piece blocks which is why I went with it. Let me know if you have other ideas.

File: core/block_render_svg.js [hunk 1]
```diff
@@ -77,10 +77,17 @@ Blockly.BlockSvg.CORNER_RADIUS = 8;
...
+ * @const
+ */
+Blockly.BlockSvg.START_HAT_HEIGHT = 15;
+/**
  * Path of the top hat's curve.
  * @const
  */
-Blockly.BlockSvg.START_HAT_PATH = 'c 30,-15 70,-15 100,0';
+Blockly.BlockSvg.START_HAT_PATH = 'c 30,-' +
+  Blockly.BlockSvg.START_HAT_HEIGHT +' 70,-' +
```
Comment (NeilFraser) [inline core/block_render_svg.js]: Indent twice more.  Space after +.
Don't worry about it, I'll fix this post-merge.

File: core/flyout.js [hunk 2]
```diff
@@ -510,6 +510,14 @@ Blockly.Flyout.prototype.reflow = function() {
         var blockXY = block.getRelativeToSurfaceXY();
         block.flyoutRect_.setAttribute('x',
             this.RTL ? blockXY.x - blockHW.width + tab : blockXY.x - tab);
+        // For hat blocks we want to shift them down by the hat height
+        // since the y coordinate is the corner, not the top of the
+        // hat.
+        var hatOffset =
+          block.startHat_ ? Blockly.BlockSvg.START_HAT_HEIGHT : 0;
```
Comment (NeilFraser) [inline core/flyout.js]: Indent twice more.
Don't worry about it, I'll fix this post-merge.

File: core/flyout.js [hunk 3]
```diff
@@ -510,6 +510,14 @@ Blockly.Flyout.prototype.reflow = function() {
...
+        if (hatOffset != 0) {
```
Comment (NeilFraser) [inline core/flyout.js]: In JS you can just say 'if (hatOffset) {' for a 3 byte win (post compile).
Don't worry about it, I'll fix this post-merge.

**Pull request merged**